### DISCUSSION TITLE
Full page components not accepting route parameters

### DISCUSSION THREAD

**Discussion by AmG2000** [livewire/livewire#3404] [7. Bugs (archived)]
I noticed this issue after I upgraded from version 2.4.4 to version 2.5.2 (because of the google cloud storage disk support for temporary file uploads). Problem is, now all my full page components that make use of route model binding now fail with a `BadMethodCallException,
Method Illuminate\Routing\Route::getMissing` does not exist.  

Here's an example of one of my components and the corresponding route definition. 

When I comment out the mount method and the Recipe variable and all its references, everything works well. 

Component class

`
class SubmitRecipeFiles extends Component
{
    use WithFileUploads, AuthorizesRequests;

    public $images = [];

    public $recipe;

    public function mount(Recipe $recipe)
    {
        $this->recipe = $recipe;
    }

    public function updatedImages()
    {
        $this->validate([
            'images' => ['required','array'],
            'images.*' => ['required', 'image', 'mimes:jpeg,jpg,png', 'max:10000'],
        ]);
    }

    public function saveImages()
    {
        $this->authorize('update',$this->recipe);
        $unsavedImages = [];
        foreach ($this->images as $image) {
            $filePath = Storage::disk('gcs')->putFile('recipeImages', $image);
            $image = new Image(['url' => Storage::disk('gcs')->url($filePath)]);
            array_push($unsavedImages, $image);
        }
        $this->recipe->images()->saveMany($unsavedImages);

        return redirect('/chef');
    }
    public function render()
    {
        return view('livewire.chef.submit-recipe-files')
        ->extends('layouts.dashboard_master')
        ->section('main');
    }
}
`

Route
`Route::get('/recipes/{recipe}/files', [\App\Http\Livewire\Chef\SubmitRecipeFiles::class, '__invoke'])->middleware('auth:chef');`

Laravel 7.22.2
php 7.3.1
livewire 2.5.2

**Comment by krokovicsmate** [comment]
@AmG2000 Your route is not required the `__invoke`, maybe thats the problem. 
`Route::get('/recipes/{recipe}/files', \App\Http\Livewire\Chef\SubmitRecipeFiles::class)->middleware('auth:chef');`

**Comment by AmG2000** [comment]
It does require the __invoke since i'm using laravel 7.

**Comment by joshhanley** [comment]
@AmG2000 it should still work ok in Laravel 7 the way @krokovicsmate suggested. The only thing that changed between 7 and 8 was removal of prefixing the controller path.

But it seem like this PR https://github.com/livewire/livewire/pull/2955 has caused your issues.

It is making use of `getMissing` which was added in Laravel 8 but no check has been put in place for Laravel 7 or earlier versions of Laravel 8 to see if the method exists first.

@ariaieboy any chance you can have a look at this? Caleb said he was going to revert your changes but it looks like he fixed the issue instead.

**Comment by AmG2000** [comment]
@joshhanley i tried @krokovicsmate 's  suggestion but it threw another error saying `function() undefined`.
But I think you're right about the `getMissing ` function since my problem started after I upgraded from version 2.4.4 to 2.5.2.
I've tried the other versions vetween 2.5.2 and 2.5.5 and they all have the same issue
